utils/build-script

```diff
@@ -63,7 +62,718 @@ def call_without_sleeping(command, dry_run=False):
+        if host_target == args.host_target:
+            # This host is the user's desired product, so honor the requested
+            # set of targets to configure/build.
+            needed_deployment_targets = args.stdlib_deployment_targets
+            if "all" in args.build_stdlib_deployment_targets:
+                needed_deployment_targets_to_build = set(
+                    needed_deployment_targets)
+            else:
+                needed_deployment_targets_to_build = set(
+                    args.build_stdlib_deployment_targets).intersect(
+                        set(args.stdlib_deployment_targets))
+        else:
+            # Otherwise, this is a host we are building as part of
+            # cross-compiling, so we only need the target itself.
+            needed_deployment_targets = [host_target]
+            needed_deployment_targets_to_build = set(needed_deployment_targets)
+
+        # Compute the lists of **CMake** targets for each use case and the SDKs
+        # to configure with.
+        #
+        # NOTE: We frequently (when building on Darwin) configure a larger
+        # set of targets than we build for by default. This is so that
+        # developers can easily build for iOS without reconfiguring.
+        self.stdlib_targets_to_build = []
+        self.targets_to_test = []
+        self.benchmark_targets_to_build = []
+        self.benchmark_targets_to_run = []
+        self.sdks_to_configure = set()
+        for deployment_target_name in needed_deployment_targets:
+            # Get the target object.
+            deployment_target = StdlibDeploymentTarget.get_target_for_name(
+                deployment_target_name)
+            if deployment_target is None:
+                diagnostics.fatal("unknown target: %r" % (
+                    deployment_target_name,))
+
+            # Add the SDK to use.
+            deployment_platform = deployment_target.platform
+            self.sdks_to_configure.add(deployment_platform.sdk_name)
+
+            # If we aren't actually building this target (only configuring
+            # it), do nothing else.
+            if deployment_target_name not in needed_deployment_targets_to_build:
+                continue
+
+            # Compute which actions are desired.
+            build = deployment_platform not in \
+                    invocation.platforms_to_skip_build
+            test = deployment_platform not in \
+                   invocation.platforms_to_skip_test
+            test_host_only = None
+            build_benchmark = build and deployment_target.supports_benchmark
+            # FIXME: Note, `build-script-impl` computed a property here
+            # w.r.t. testing, but it was actually unused.
+
+            # For platforms which normally require a connected device to
+            # test, the default behavior is to run tests that only require
+            # the host (i.e., they do not attempt to execute).
+            if deployment_platform.is_darwin and \
```

Comment:
We only have comparable checks in two places: 
- if a target has darwin symbols (i.e. whether or not we should `dsymutil` -- but we could generalise that as an `executable_format` and check for `Mach-O` instead; it's technically more accurate and that's useful information for cross-compilers anyway. I built `llvm-dsymutil` for linux, and if I had any Mach-Os on there I needed to strip, I'd be using it.)
- Whether or not to lipo. Could also go under the `Mach-O` heading.